debian/mariadb-server-10.6.preinst

```diff
@@ -119,6 +119,16 @@ then
 
 fi
 
+# If there is no debian-*.flag, and no version was detected, but a file that
+# indicated MySQL 8.0 is found, assume so and trigger data directory moval.
+if [ -z "$found_version" ] &&
+   [ -z "$(find $mysql_datadir/debian-*.flag 2> /dev/null)" ] &&
+   [ -f "$mysql_datadir/undo_001" ]
```

Comment:
As this was bit tricky to understand please add comment about `undo_001` why it's used to determine MySQL 8.0. I found it from docs that it's created default.